src/software/ai/hl/stp/play/stop_play_test.cpp

```diff
@@ -10,36 +12,225 @@
+            {
+                yield();
+            }
+            // TODO: (#1882) implement robots slow down when responding to stop command
+            // robotsSlowDown(1.5, world_ptr, yield);
+            robotsAvoidBall(0.5, world_ptr, yield);
+        }};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+// TODO: #1948 fix warning message "Navigator's path manager could not find a path"
+TEST_F(StopPlayTest, test_stop_play_ball_in_front_of_enemy_defense_area)
+{
+    setBallState(BallState(Point(3, 0), Vector(0, 0)));
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(-4.5, 2), Point(0, 0.3), Point(0.3, 0), Point(0, -0.3), Point(-0.3, 0),
+         Point(0.2, 0.2)}));
+
+    std::vector<ValidationFunction> terminating_validation_functions = {};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            // Wait 2 seconds for robots that start too close to the ball to move away
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(2))
+            {
+                yield();
+            }
+            // TODO: (#1882) implement robots slow down when responding to stop command
+            // robotsSlowDown(1.5, world_ptr, yield);
+            robotsAvoidBall(0.5, world_ptr, yield);
+        }};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+// TODO: (#1948) Re-enable test once bug is fixed, fix warning and robot positioning
+// This test is disabled due to a bug that causes Robot 2 to move too close to the ball.
+// Robots' positioning was not good, two robots were on the wrong side of the ball.
+// This test produced the warning, "Navigator's path manager could not find a path"
+TEST_F(StopPlayTest, DISABLED_test_stop_play_ball_in_front_of_friendly_defense_area)
+{
+    setBallState(BallState(Point(-3, 0), Vector(0, 0)));
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(-4.5, 2), Point(0, 3), Point(-1, -1), Point(0, 0), Point(-1, 0),
+         Point(2, 2)}));
+
+    std::vector<ValidationFunction> terminating_validation_functions = {};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            // Wait 2 seconds for robots that start too close to the ball to move away
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(2))
+            {
+                yield();
+            }
+            // TODO: (#1882) implement robots slow down when responding to stop command
+            // robotsSlowDown(1.5, world_ptr, yield);
+            robotsAvoidBall(0.5, world_ptr, yield);
+        }};
```

Comment:
We can make this a class variable in StopPlayTest since we seem to reuse it a lot. We should also give it a name like stop_play_rules

Reply:
Fixed